frontend/lib/src/components/widgets/DataFrame/columns/cells/LinkCell.tsx

```diff
@@ -0,0 +1,174 @@
+ */
+import * as React from "react"
+
+import {
+  CustomCell,
+  CustomRenderer,
+  getMiddleCenterBias,
+  measureTextCached,
+  blend,
+  GridCellKind,
+} from "@glideapps/glide-data-grid"
+import UriOverlayEditor from "./UriOverlayEditor"
+
+const UNDERLINE_OFFSET = 5
+
+export interface LinkCellProps {
+  readonly kind: "link-cell"
+  readonly href: string
+  readonly displayText?: string
+}
+
+export type LinkCell = CustomCell<LinkCellProps>
+
+function onClickSelect(
+  e: Parameters<NonNullable<CustomRenderer<LinkCell>["onSelect"]>>[0]
+): string | undefined {
+  const canvas = document.createElement("canvas")
+  const ctx = canvas.getContext("2d", { alpha: false })
+  if (ctx === null) return
+
+  const { posX: hoverX, bounds: rect, cell, theme } = e
+  const font = `${theme.baseFontStyle} ${theme.fontFamily}`
+  ctx.font = font
+
+  const { href, displayText } = cell.data
+
+  const rectHoverX = rect.x + hoverX
+
+  const textWidth =
+    ctx.measureText(displayText || href).width +
+    theme.cellHorizontalPadding * 2
+
+  const isHovered = rectHoverX > rect.x && rectHoverX < rect.x + textWidth
+
+  if (isHovered) {
+    return href
+  }
+
+  return undefined
+}
+
+export const linkCellRenderer: CustomRenderer<LinkCell> = {
+  draw: (args, cell) => {
+    const { ctx, rect, theme, hoverX = -100, highlighted } = args
+    const { href, displayText } = cell.data
+    if (href === undefined) return
+
+    const displayValue = displayText || href
+
+    const xPad = theme.cellHorizontalPadding
+
+    const drawX = rect.x + xPad
+
+    const rectHoverX = rect.x + hoverX
+
+    const font = `${theme.baseFontStyle} ${theme.fontFamily}`
+    ctx.font = font
+    const middleCenterBias = getMiddleCenterBias(ctx, font)
+    const drawY = rect.y + rect.height / 2 + middleCenterBias
+
+    const isHovered = rectHoverX > rect.x && rectHoverX < rect.x + rect.width
+
+    if (isHovered) {
+      // draw the underline only when the cell is hovered
+      const metrics = measureTextCached(displayValue, ctx, font)
+
+      ctx.moveTo(drawX, Math.floor(drawY + UNDERLINE_OFFSET) + 0.5)
+      ctx.lineTo(
+        drawX + metrics.width,
+        Math.floor(drawY + UNDERLINE_OFFSET) + 0.5
+      )
+
+      ctx.strokeStyle = theme.linkColor
+      ctx.stroke()
+
+      ctx.fillStyle = highlighted
+        ? blend(theme.accentLight, theme.bgCell)
+        : theme.bgCell
+      ctx.fillText(displayValue, drawX - 1, drawY)
+      ctx.fillText(displayValue, drawX + 1, drawY)
+
+      ctx.fillText(displayValue, drawX - 2, drawY)
+      ctx.fillText(displayValue, drawX + 2, drawY)
+    }
+
+    ctx.fillStyle = theme.linkColor
+    ctx.fillText(displayValue, drawX, drawY)
+    ctx.closePath()
+
+    return true
+  },
+  isMatch: (c): c is LinkCell =>
+    (c.data as LinkCellProps).kind === "link-cell",
+  kind: GridCellKind.Custom,
+  measure: (ctx, cell, theme) => {
+    const { href, displayText } = cell.data
+    if (href === undefined) return 0
```

Comment:
nit: I think we either need to also check displayText here or just simplify it to:

```
return (
      ctx.measureText(displayText || href || "").width +
      theme.cellHorizontalPadding * 2
    )
 ```

Reply:
I believe this got addressed, but let me know if not:

```
measure: (ctx, cell, theme) => {
    const { href, displayText } = cell.data
    if (isNullOrUndefined(href)) return 0

    return (
      ctx.measureText(displayText || href).width +
      theme.cellHorizontalPadding * 2
    )
  },
  ```

Reply:
Yep, I think that is fine 👍 Theoretically, there could be a situation with the LinkCell where `href` is null and `displayText` is set. But it's not really possible in our column implementation.